File: m2-h2/src/main/java/ru/yegorpilipenko/otus/spring/m2h2/repository/BookRepository.java
package ru.yegorpilipenko.otus.spring.m2h2.repository;

import org.springframework.data.mongodb.repository.MongoRepository;
import ru.yegorpilipenko.otus.spring.m2h2.document.Book;

import javax.annotation.Nonnull;

public interface BookRepository extends MongoRepository<Book, String>, ExtendedBookRepository {

    int removeById(@Nonnull String id);

}

File: m2-h1/src/main/kotlin/ru/yegorpilipenko/otus/spring/m2h1/repository/BookRepository.kt
package ru.yegorpilipenko.otus.spring.m2h1.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import ru.yegorpilipenko.otus.spring.m2h1.entity.Book

@Repository
interface BookRepository : JpaRepository<Book, Long> {

    fun removeById(id: Long): Int
}